### ISSUE TITLE
Consider using d1 as storage backend rather than kv

### ISSUE THREAD

**Issue by Enter-tainer** [yunsii/cf-worker-status-page-pro#6]
https://developers.cloudflare.com/d1/platform/pricing/

cloudflare launches d1 recently. It is a cloud native sqlite solution. The free tier allows 100000 writes per day. This is 100x more than kv.

I think we can create a table with only 2 columns, one for key, the other one for value. So we can still expose a kv interface. So we only need change https://github.com/yunsii/cf-worker-status-page-pro/blob/master/src/worker/_helpers/store.ts

**Comment by Enter-tainer** [comment]
https://developers.cloudflare.com/d1/platform/limits/ One thing to notice is that the max row size is 1MB. This is smaller than kv.

**Comment by yunsii** [comment]
Furthermore, support to switch storage solution (KV/D1/R2) is a better solution, I migrated the storage to R2 in my company 😂 and it is low priority for now, but PR is welcome.

**Comment by Enter-tainer** [comment]
r2 is pretty cool. It also has large r/w quota. And it has a much higher limit for object size. 

>  I migrated the storage to R2 in my company 😂

Just being curious, will you open source relevant code here?

**Comment by yunsii** [comment]
Yes, I will migrate to [Pages](https://developers.cloudflare.com/pages/) firstly 😏

**Comment by Enter-tainer** [comment]
that's fascinating! waiting for your code🥰

**Comment by yunsii** [comment]
After I tried, Cloudflare Pages [does not support Cron Triggers](https://community.cloudflare.com/t/cron-triggers-in-pages/430734) for now.

I think it is so complicated if used with another separate worker, so I decided to wait for its support 😂 And try to support [https://github.com/yunsii/cf-worker-status-page-pro/issues/5] firstly.

**Comment by JoyceBabu** [comment]
You should enable Smart Placement when using D1, or else the requests will be painfully slow when the worker node is at a different location from the D1 node.
